contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1187,3 +1194,127 @@ babelfish_integrity_checker(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(true);
 }
+
+Datum
+int_power(PG_FUNCTION_ARGS)
+{
+	int64	arg1 = PG_GETARG_INT64(0);
+	int64	arg2 = PG_GETARG_INT64(1);
+	float8	result;
+
+	result = DatumGetFloat8(DirectFunctionCall2(dpow, Float8GetDatum((float8) arg1),Float8GetDatum((float8) arg2)));
+
+	if (result < 0)
+    	result = ceil(result);
+	else
+    	result = floor(result);
+
+	/* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT64(result)))
+		ereport(ERROR,
+			(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+			errmsg("integer out of range")));
+
+	PG_RETURN_INT64((int64)result);
+ 
+}
+
+Datum
+int_radians(PG_FUNCTION_ARGS)
+{
+	int64	arg1 = PG_GETARG_INT64(0);
+	float8	result;
+
+	result = DatumGetFloat8(DirectFunctionCall1(radians, Float8GetDatum((float8) arg1)));
+
+	PG_RETURN_INT64((int64)result);
+	 
+}
+
+Datum
+BIGINT_degrees(PG_FUNCTION_ARGS)
+{
+	int64	arg1 = PG_GETARG_INT64(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+    	result = ceil(result);
+	else
+    	result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT64(result)))
+		ereport(ERROR,
+				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+				errmsg("integer out of range")));
+
+    PG_RETURN_INT64((int64)result);
+}
+
+Datum
+int_degrees(PG_FUNCTION_ARGS)
+{
+	int32	arg1 = PG_GETARG_INT32(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+    	result = ceil(result);
+	else
+    	result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT32(result)))
+		ereport(ERROR,
+				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+				errmsg("integer out of range")));
+
+    PG_RETURN_INT32((int32)result);
+}
+
+Datum
+smallint_degrees(PG_FUNCTION_ARGS)
+{
+	int16	arg1 = PG_GETARG_INT16(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+    	result = ceil(result);
+	else
+    	result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT16(result)))
+		ereport(ERROR,
+				(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
+				errmsg("integer out of range")));
+
+    PG_RETURN_INT16((int16)result);
+}
+
+Datum
+tinyint_degrees(PG_FUNCTION_ARGS)
+{
+	int16	arg1 = PG_GETARG_INT16(0);
+	float8	result;
+	 
+	result = DatumGetFloat8(DirectFunctionCall1(degrees, Float8GetDatum((float8) arg1)));
+
+	if (result < 0)
+    	result = ceil(result);
+	else
+    	result = floor(result);
+
+	 /* Range check */
+	if (unlikely(isnan(result) || !FLOAT8_FITS_IN_INT16(result)))
```

Comment:
Here, you may need to check whether it fits int 1Byte, which is definition of tinyint